src/utilities/axisBumpFactory.ts

```diff
@@ -0,0 +1,68 @@
+// Map<YAxis, Map<NodeId, offset>>
+type AxisBumps = Map<number, Map<string, number> | undefined>
+
+type SetBumpParameters = {
+  axis: number,
+  nodeId: string,
+  offset: number,
+}
+
+type RemoveBumpParameters = {
+  axis: number,
+  nodeId: string,
+}
+
+type BumpFactory = {
+  get: (axis: number) => number,
+  set: (value: SetBumpParameters) => void,
+  delete: (value: RemoveBumpParameters) => void,
+  bump: (axis: number) => number,
+  clear: () => void,
+}
+
+export function axisBumpFactory(): BumpFactory {
```

Comment:
maybe `yAxisOffsetFactory`. I'm not particularly offended by `bump` nomenclature, but as you alluded it might not be the most descriptive in the long run.